src/admin/settings/settings.controller.js

```diff
@@ -0,0 +1,127 @@
+            $("#settingsFileForm").ajaxSubmit({
+                success: function () {
+                    LoadingModal.close();
+                    ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.saved')),
+                        i18nService.getMessage('admin.success')).open();
+                    $scope.platformSettings = PlatformSettingsFactory.get();
+                },
+                error: function (response) {
+                    LoadingModal.close();
+                    ModalWindow(angular.toJson(response.status + ": " + response.statusText),
+                        i18nService.getMessage('admin.settings.error')).open();
+                }
+            });
+        };
+
+        $scope.uploadFileLocation = function () {
+            $http({method:'POST', url: ServerService.formatURL('module/admin/api/settings/platform/location'), params: {location: this.location}})
+                .success( function () {
+                    LoadingModal.close();
+                    ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.saved')), i18nService.getMessage('admin.success')).open();
+                })
+                .error( function () {
+                   LoadingModal.close();
+                   ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.error.location')),
+                       i18nService.getMessage('admin.settings.error')).open();
+                });
+        };
+
+        $scope.saveAll = function () {
+            LoadingModal.open();
+            $http.post(ServerService.formatURL('module/admin/api/settings/platform/list'), $scope.platformSettings.settingsList)
+                .success( function () {
+                    LoadingModal.close();
+                    ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.saved')), i18nService.getMessage('admin.success')).open();
+                })
+                .error( function () {
+                    LoadingModal.close();
+                    ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.error.location')),
+                        i18nService.getMessage('admin.settings.error')).open();
+                });
+        };
+
+        $scope.exportConfig = function () {
+            $http.get(ServerService.formatURL('module/admin/api/settings/platform/export'))
+            .success(function () {
+                window.location.replace(ServerService.formatURL("module/admin/api/settings/platform/export"));
+            })
+            .error( function () {
+                LoadingModal.close();
+                ModalWindow(angular.toJson(i18nService.getMessage('admin.settings.error.export')),
+                    i18nService.getMessage('admin.settings.error')).open();
+            });
+        };
+
+        $scope.addValue = function(key, value) {
+            if ($scope.comboboxValues[key].indexOf(value) === -1) {
+                $scope.comboboxValues[key].push(value);
```

Comment:
This is out of scope of this ticket (slightly) - but the better UX for this field input would be for the option field to set to "" after the option is added, and the combo box to be set to the new option (so that the user can see something changed)

If you want to approach this functionality, please comment here — otherwise I will create a new ticket just for this item